include/openPMD/ParticleSpecies.hpp

```diff
@@ -62,11 +62,11 @@ namespace traits
 
             auto& np = ret.particlePatches["numParticles"];
             auto& npc = np[RecordComponent::SCALAR];
-            npc.resetDataset(Dataset(Datatype::UINT64, {0}));
+            npc.resetDataset(Dataset(Datatype::UINT64, {1}));
```

Comment:
Why do we set an extend of `1` here already? Doesn't this imply a first entry already? What if someone does not write data / writes exactly one entry later on? Just wondering about corner cases.

Reply:
ADIOS does not handle datasets without extent well. The default here is just used to enforce standard compliance and create deterministic behaviour across backends.
To address your concern about corner cases: [```PatchRecordComponent::resetDataset()```](https://github.com/openPMD/openPMD-api/blob/0.3.1-alpha/src/backend/PatchRecordComponent.cpp#L33-L42) should be used by users that do want to write actual data. It will be replaced when writing production data.
If the user does *not* write dataset values ever, the created dataset will contain one potentially [default](https://github.com/openPMD/openPMD-api/blob/414f2ddbdd1e0542d6f82aa32885e3adbfe7eef4/src/RecordComponent.cpp#L107-L121)-initalized value.